src/clj/puppetlabs/services/jruby/jruby_puppet_internal.clj

```diff
@@ -95,44 +91,55 @@
   [ruby-load-path :- [schema/Str]]
   (cons ruby-code-dir ruby-load-path))
 
-(defn prep-scripting-container
-  [scripting-container ruby-load-path gem-home]
-  ; Note, this behavior should remain consistent with new-main
-  (doto scripting-container
+(schema/defn ^:always-validate get-compile-mode :- RubyInstanceConfig$CompileMode
+  [config-compile-mode :- jruby-schemas/SupportedJRubyCompileModes]
+  (case config-compile-mode
+    :jit RubyInstanceConfig$CompileMode/JIT
+    :force RubyInstanceConfig$CompileMode/FORCE
+    :off RubyInstanceConfig$CompileMode/OFF))
+
+(schema/defn ^:always-validate init-jruby-config :- jruby-schemas/ConfigurableJRuby
+  "Applies configuration to a JRuby... thing.  See comments in `ConfigurableJRuby`
+  schema for more details."
+  [jruby-config :- jruby-schemas/ConfigurableJRuby
```

Comment:
FWIW, Jeff and I had talked about doing this as a common helper that called the same on either type regardless of the grossness of not having a common Java interface available to enforce the contract.  I think we chose not to do it then because you were concerned about not having that common Java interface.  I'm still okay with doing it this way for the sake of keeping the initialization common between the `ScriptingContainer` and JRuby `Main` for the CLI use cases.